src/suit_decoder.c

```diff
@@ -184,39 +132,34 @@ static int get_component_id_str(struct zcbor_string *out_component_id,
 	const struct SUIT_Component_Identifier *component_id)
 {
 	const struct zcbor_string *first_bstr = &component_id->_SUIT_Component_Identifier_bstr[0];
-	size_t payload_len = 0;
-
-	int first_bstr_len = header_len(first_bstr->len, &first_bstr->value[0]);
+	int first_bstr_len = header_len(first_bstr->len, &first_bstr->value[0], ZCBOR_MAJOR_TYPE_BSTR);
 
 	if (first_bstr_len < 0) {
 		return SUIT_ERR_DECODING;
 	}
 
-	int extra_header_len = header_len(component_id->_SUIT_Component_Identifier_bstr_count, &(first_bstr->value[0]) - first_bstr_len);
+	int extra_header_len = header_len(component_id->_SUIT_Component_Identifier_bstr_count, &(first_bstr->value[0]) - first_bstr_len, ZCBOR_MAJOR_TYPE_LIST);
 
 	if (extra_header_len < 0) {
 		return SUIT_ERR_DECODING;
 	}
 
-	/* Do not allow empty lists as components ID values (already verified by CDDL). */
-
-	for (size_t i = 0; i < component_id->_SUIT_Component_Identifier_bstr_count; i++) {
-		/* Do not allow empty byte strings as component ID values. */
-		if (component_id->_SUIT_Component_Identifier_bstr[i].len == 0) {
-			return SUIT_ERR_DECODING;
-		}
-
-		int chunk_len = header_len(component_id->_SUIT_Component_Identifier_bstr[i].len, &component_id->_SUIT_Component_Identifier_bstr[i].value[0]);
+	/* Do checks on component ID parts. */
+	for (int i = 0; i < component_id->_SUIT_Component_Identifier_bstr_count; i++) {
+		const struct zcbor_string *comp_bstr = &component_id->_SUIT_Component_Identifier_bstr[i];
 
-		if (chunk_len < 0) {
+		if (header_len(comp_bstr->len, &comp_bstr->value[0], ZCBOR_MAJOR_TYPE_BSTR) < 0
+				|| component_id->_SUIT_Component_Identifier_bstr[i].len == 0) {
 			return SUIT_ERR_DECODING;
 		}
-		payload_len += chunk_len;
-		payload_len += component_id->_SUIT_Component_Identifier_bstr[i].len;
 	}
 
 	out_component_id->value = first_bstr->value - extra_header_len - first_bstr_len;
-	out_component_id->len = extra_header_len + payload_len;
+	out_component_id->len = component_id->_SUIT_Component_Identifier_bstr[
```

Comment:
You may define a helper variable `last_bstr` to make it a little shorter :slightly_smiling_face: